coreneuron/utils/randoms/nrnran123.cu

```diff
@@ -157,6 +220,8 @@ nrnran123_State* nrnran123_newstream3(uint32_t id1,
     s->c.v[3] = id2;
     nrnran123_setseq(s, 0, 0);
     {
+        // TODO: can I assert something useful about the instance count going
+        // back to zero anywhere? Or that it is zero when some operations happen?
```

Comment:
Answered in previous comment.

@nrnhines : How the ran123 streams allocated with `nrnran123_newstream3` inside `bbcore_read()` should be allocated? any thoughts?

Reply:
lets keep this as a separate issue. https://github.com/BlueBrain/nmodl/issues/383 would help to implement this easily.